lib/vsc/filesystem/oceanstor.py

```diff
@@ -405,3 +416,153 @@ def list_filesets(self, devices=None, filesystemnames=None, filesetnames=None, u
+                # Check NFS mount server and filesystem name
+                mount_device = fs[self.localfilesystemnaming.index('device')]
+                mount_ip, fs_path = mount_device.split(':', 1)
+
+                # Determine if filesystem is a known filesystem in OceanStor
+                if mount_ip in self.nfs_ips:
+                    fs_name = os.path.basename(fs_path)
+                    try:
+                        fs_oceanstor_name = self.oceanstor_filesystems[fs_name]['name']
+                    except KeyError as err:
+                        errmsg = "NFS mount '%s' served from OceanStor has an unkown filesystem '%s' to the REST API"
+                        self.log.raiseException(errmsg % (mount_device, fs_name), OceanStorOperationError)
+
+            # Add filesystem name in OceanStor to all mounts (even if None)
+            fs.append(fs_oceanstor_name)
+
+    def make_fileset(self, new_fileset_path, fileset_name=None, parent_fileset_name=None, afm=None,
+                     inodes_max=None, inodes_prealloc=None):
+        """
+        Create a new fileset in a NFS mounted filesystem from OceanStor
+
+        - The name of the dtree fileset and the folder where it is mounted always share the same name.
+        - Dtree filesets cannot be nested (parent_fileset_name is ignored)
+        - Dtree filesets can be created at specific path inside the NFS mount (i.e. filesystem),
+          but this information cannot be retrieved back from the OceanStor API (list_filesets()).
+          Therefore, all filesets in a common filesystem must have unique names.
+
+        @type new_fileset_path: string with the full path in the local system of the new fileset
+        @type fileset_name: string with the name of the new fileset
+                            (if not None, fileset_name is appended to new_fileset_path)
+        """
+        # Unsupported features
+        del parent_fileset_name
+        del afm
+        del inodes_max
+        del inodes_prealloc
+
+        dt_fullpath = self._sanity_check(new_fileset_path)
+
+        if fileset_name is None:
+            # Use name of last folder as name of dtree fileset in OceanStor
+            fileset_name = os.path.basename(dt_fullpath)
+        else:
+            # Append the fileset name to the given path
+            dt_fullpath = os.path.join(dt_fullpath, fileset_name)
+            dt_fullpath = self._sanity_check(dt_fullpath)
+
+        # Check existence of path in local filesystem
+        if self.exists(dt_fullpath):
+            errmsg = "Path of new fileset '%s' is validated as '%s' but it already exists."
+            self.log.raiseException(errmsg % (new_fileset_path, dt_fullpath), OceanStorOperationError)
+
+        dt_parentdir = os.path.dirname(dt_fullpath)
+        if not self.exists(dt_parentdir):
+            errmsg = "Parent directory '%s' of new fileset '%s' does not exist. It will not be created automatically."
```

Comment:
we can do that with the api actually,

Reply:
Yes, or even with `mkdir` since the filesystem has to be mounted. But this follows the same behavior of other modules in `vsc.filesystem`. So I took it as policy.

Reply:
It's not documented but you can let the API do it ;)

But let's keep to the same behaviour as the other modules.